### PR TITLE
Add function to port forward to local stream to support VSCode remote ssh target.

### PR DESCRIPTION
**Pull request by puppywang**
VSCode remote ssh plugin need to create port forward between local port and remote unix socket, which is not yet support by current version of dropbear, this patch add a simple impl of port forwarding to local stream, only test on centos, debian x64, should work on any other linux variant. Any advice is welcome, thanks.

### CODE REVIEW

File: netio.c [hunk 1]
```diff
@@ -225,6 +225,61 @@ struct dropbear_progress_connection *connect_remote(const char* remotehost, cons
...
+	list_append(&ses.conn_pending, c);
+
+#if DROPBEAR_FUZZ
+	if (fuzz.fuzzing) {
+		c->errstring = m_strdup("fuzzing connect_streamlocal always fails");
+		return c;
+	}
+#endif
+
+	if (strlen(localpath) > sizeof(sunaddr->sun_path)) {
```
Comment (mkj) [inline netio.c:253]: This should be `>=` not `>`, need space for the null terminator

**Review by mkj — CHANGES_REQUESTED** [review]
Thanks, this seems useful. I've added a few comments inline for changes.

One other thing that needs changing is `connect_try_next()`. It should skip the `fastopen` path when it's `AF_UNIX` family, and according to the `connect(2)` manpage it should be testing `EAGAIN` rather than `EINPROGRESS` for unix sockets. If you'd prefer I can make those changes later.

I guess "tcpfwd" isn't an accurate file name any more - maybe the files should be renamed to svr-fwd.c, cli-fwd.c, fwd.h. Again, I'm happy to change that separately.

**Comment by puppywang** [comment]
Thanks for your reply, I will fix those issues later.

**Comment by puppywang** [comment]
All issues resolved.

**Comment by mkj** [comment]
Thanks, sorry for the delay getting this merged

**Pull request merged**